test/utils/UtilsSpec.scala

```diff
@@ -57,4 +60,49 @@ class UtilsSpec extends SpecBase {
       referrerUrl(None) mustBe Option(s"$path")
     }
   }
+
+  "isEqualOrAfter" must {
+    "return true when date is equal to or after cutOffDate" in new Setup {
+      Utils.isEqualOrAfter(date03, cutOffDate) shouldBe true
+      Utils.isEqualOrAfter(date04, cutOffDate) shouldBe true
+    }
+
+    "return false when date is before cutOffDate" in new Setup {
+      Utils.isEqualOrAfter(date02, cutOffDate) shouldBe false
+    }
+  }
+
+  "isEqualOrBefore" must {
+    "isEqualOrBefore should return true when date is equal to or before cutOffDate" in new Setup {
+      Utils.isEqualOrBefore(date03, cutOffDate) shouldBe true
+      Utils.isEqualOrBefore(date02, cutOffDate) shouldBe true
+    }
+
+    "isEqualOrBefore should return false when date is after cutOffDate" in new Setup {
+      Utils.isEqualOrBefore(date04, cutOffDate) shouldBe false
+    }
+  }
+
+  "firstDayOfPastNthMonth" must {
+    "return the first day of the month 3 months ago" in new Setup {
+      Utils.firstDayOfPastNthMonth(date03, numberOfMonths) shouldBe date01
+    }
+  }
+
+  trait Setup {
+    val YEAR_2024: Int = 2024
+    val MONTH_06: Int = 6
```

Comment:
Could you please add a line break before this line ?

Also make use of line break as mentioned for below lines

Line break after line 95
Line break after line 99
Line break after line 104

Above is to group them that makes more readable.